### PR TITLE
chore(server): clean up server code

### PR DESCRIPTION
**Pull request by emilyrohrbough**
- Remove dead code left over from migrating server code to the data-context package during the 10.x uplift.
- update run-e2e.js -> run-e2e.ts
- fix types

### CODE REVIEW

File: packages/server/lib/modes/run-e2e.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import type { LaunchArgs } from '@packages/types'
+
+export const run = (options: LaunchArgs, loading: Promise<void>) => {
+  return require('./run').run(options, loading)
```
Comment (lmiller1990) [inline packages/server/lib/modes/run-e2e.ts:4]: I wonder if we want to start moving towards ESM more?

```js
import { run as runE2E } from './run'

export const run = (options: LaunchArgs, loading: Promise<void>) => {
  return runE2E(options, loading)
}
```

**Review by lmiller1990 — APPROVED** [review]
Love PRs that delete more than they add.

File: packages/server/lib/modes/run-e2e.ts (hunk 1 above)
Comment (emilyrohrbough) [reply packages/server/lib/modes/run-e2e.ts:4]: 🤷🏻‍♀️ I don't have a strong pref, just copied run-ct 😄 https://github.com/cypress-io/cypress/blob/clean-up-server/packages/server/lib/modes/run-ct.ts

**Review by mschile — APPROVED** [review]

**Review by lmiller1990 — APPROVED** [review]

File: packages/server/test/integration/cypress_spec.js [hunk 2]
```diff
@@ -20,7 +20,6 @@ const argsUtil = require(`../../lib/util/args`)
 const { fs } = require(`../../lib/util/fs`)
 const ciProvider = require(`../../lib/util/ci_provider`)
```
Comment (lmiller1990) [inline packages/server/test/integration/cypress_spec.js:21]: I notice `  options.browser = options.browser || 'electron'` got lost in `run-ct.ts`

Is this now un-needed?

Comment (emilyrohrbough) [reply packages/server/test/integration/cypress_spec.js:21]: It's the default here: https://github.com/cypress-io/cypress/pull/23002/files#diff-34201b19e236c2f1e4512cce84d30f85425eaa6f293a0b57025f038b59190bf7R20

Comment (lmiller1990) [reply packages/server/test/integration/cypress_spec.js:21]: ^ link doesn't seem to go to a specific line but I will take your word for it :+1: approved.

**Review by lmiller1990 — APPROVED** [review]

**Pull request merged**